File: InterviewProblems/src/utils/scala/Node1.scala
package utils.scala

/**
 * Created with IntelliJ IDEA.
 * User: Balaji Athreya
 * Date: 9/21/13
 * Time: 7:58 PM
 * To change this template use File | Settings | File Templates.
 */
class Node1(v : Int){
  var parent : Node1 = null
  var left : Node1 = null
  var right : Node1 = null
  var value = v

  def setLeft(l : Node1) {left = l}
  def setRight(r : Node1) {right = r}

  override def toString = value.toString
}

object Node1{


  def buildTree : (Node1,Node1) = {
    val root = new Node1(1)
    val left = new Node1(2)
    val right = new Node1(3)
    root.left = left
    root.right = right
    left.parent = root
    right.parent = root

    val left1 = new Node1(4)
    val left2 = new Node1(5)
    left.left = left1
    left.right = left2
    left1.parent = left
    left2.parent = left

    val right1 = new Node1(6)
    val right2 = new Node1(7)
    right.left = right1
    right.right = right2
    right1.parent = right
    right2.parent = right

    (root, right1)
  }
}

File: InterviewProblems/src/utils/java/Node1.java
package utils.java;

/**
 * Created with IntelliJ IDEA.
 * User: Balaji Athreya
 * Date: 9/21/13
 * Time: 8:16 PM
 * To change this template use File | Settings | File Templates.
 */
public class Node1 {
    private int value;
    private Node1 left = null;
    private Node1 right = null;

    private Node1 parent = null;

    public Node1(int v){
        value = v;
    }

    public String toString(){
        return ""+value;
    }

    public Node1 getLeft() {
        return left;
    }

    public void setLeft(Node1 left) {
        this.left = left;
    }

    public Node1 getRight() {
        return right;
    }

    public void setRight(Node1 right) {
        this.right = right;
    }

    public Node1 getParent() {
        return parent;
    }

    public void setParent(Node1 parent) {
        this.parent = parent;
    }



    public static Node1 buildTree(){
        Node1 root = new Node1(1);
        Node1 left = new Node1(2);
        Node1 right = new Node1(3);
        root.setLeft(left);
        root.setRight(right);
        left.setParent(root);
        right.setParent(root);

        Node1 left1 = new Node1(4);
        Node1 left2 = new Node1(5);
        left.setLeft(left1);
        left.setRight(left2);
        left1.setParent(left);
        left2.setParent(left);

        Node1 right1 = new Node1(6);
        Node1 right2 = new Node1(7);
        right.setLeft(right1);
        right.setRight(right2);
        right1.setParent(right);
        right2.setParent(right);

        return root;

    }

}
